lib/matplotlib/backend_bases.py

```diff
@@ -2298,32 +2297,32 @@ def get_window_title(self):
             return self.manager.get_window_title()
 
     def set_window_title(self, title):
-        """
-        Set the title text of the window containing the figure.  Note that
-        this has no effect if there is no window (e.g., a PS backend).
+        """Set the title text of the window containing the figure.
+
+        This has no effect if there is no window (e.g., a PS backend).
         """
         if hasattr(self, "manager"):
             self.manager.set_window_title(title)
 
     def get_default_filename(self):
-        """
-        Return a string, which includes extension, suitable for use as
-        a default filename.
-        """
-        default_basename = self.get_window_title() or 'image'
-        default_basename = default_basename.replace(' ', '_')
+        """Return a suitable default filename, including the extension.
+        """
+        default_basename = self.get_window_title() or "image"
+        # Characters to be avoided in a NT path:
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+        removed_chars = \
+            {"posix": " /\0", "nt": r'<>:"/\|?*\0'}.get(os.name, "_")
```

Comment:
Even if it is technically ok it is still bad practice (see http://www2.stat.duke.edu/~rcs46/lectures_2015/01-markdown-git/slides/naming-slides/naming-slides.pdf slides from an R lecture, but the point stands).  We should not by default be suggesting file names with spaces in them.  We should also not suggest different default file names between windows and *nix.

I would not support sanitizing direct input from the user (ex the input to `savefig`).  If users want to shoot them selves in their foot, that is their business, but we should not be shooting their feet for them with the defaults.